src/ripple/overlay/impl/Message.cpp

```diff
@@ -41,18 +42,53 @@ Message::Message (::google::protobuf::Message const& message, int type)
 
     mBuffer.resize (headerBytes + messageBytes);
 
-    auto ptr = mBuffer.data();
+    auto set_header = [](uint8_t *ptr, decltype(messageBytes) messageBytes, int type, bool compressed=false,
+            uint8_t compr_algorithm=ripple::compression::Algorithm::LZ4) {
+        uint8_t compression = (compressed?0xF0:0) & (0x80 | (compr_algorithm << 4));
+        *ptr++ = static_cast<std::uint8_t>(((messageBytes >> 24) | compression) & 0xFF);
```

Comment:
Is the way we are using the flag here to distinguish compressed/uncompressed data documented anywhere (I didn't see it, but I may have missed it). If not, we need to document it.